crates/ef-testing/src/evm_sequencer/setup.rs

```diff
@@ -0,0 +1,246 @@
+use blockifier::abi::abi_utils::{
+    get_erc20_balance_var_addresses, get_storage_var_address, get_uint256_storage_var_addresses,
+};
+use blockifier::state::state_api::{State, StateResult};
+use reth_primitives::{Address, Bytes};
+use revm_primitives::U256;
+use starknet_api::core::Nonce;
+use starknet_api::hash::StarkFelt;
+use starknet_api::StarknetApiError;
+
+use super::constants::{
+    CONTRACT_ACCOUNT_CLASS_HASH, EOA_CLASS_HASH, FEE_TOKEN_ADDRESS, KAKAROT_ADDRESS,
+    PROXY_CLASS_HASH,
+};
+use super::types::FeltSequencer;
+use super::utils::{
+    class_hash_to_starkfelt, compute_starknet_address, contract_address_to_starkfelt,
+    split_bytecode_to_starkfelt, split_u256,
+};
+use super::KakarotSequencer;
+
+pub trait InitializeEvmState {
+    fn initialize_contract(
+        &mut self,
+        evm_address: &Address,
+        bytecode: &Bytes,
+        nonce: U256,
+        storage: Vec<(U256, U256)>,
+    ) -> StateResult<()>;
+
+    fn fund(&mut self, evm_address: &Address, balance: U256) -> StateResult<()>;
+}
+
+impl InitializeEvmState for KakarotSequencer {
+    fn initialize_contract(
+        &mut self,
+        evm_address: &Address,
+        bytecode: &Bytes,
+        nonce: U256,
+        evm_storage: Vec<(U256, U256)>,
+    ) -> StateResult<()> {
+        let nonce = StarkFelt::from(TryInto::<u128>::try_into(nonce).map_err(|err| {
+            StarknetApiError::OutOfRange {
+                string: err.to_string(),
+            }
+        })?);
+        let starknet_address = compute_starknet_address(evm_address);
+        let evm_address = Into::<FeltSequencer>::into(*evm_address).into();
+
+        let mut storage = vec![
+            (
+                get_storage_var_address("evm_address", &[]).unwrap(), // safe unwrap: var is ASCII
```

Comment:
can't we just use `?` and possibly update the signature of `initialize` to return `Result`?
This wouldn't shock me to have this function possibly failing, if for example one tries to initialize two times the same address

Reply:
`get_storage_var_address` just returns the address of the storage var so it shouldn't fail, actually since we are still on an old commit of blockifier it still returns an error but this has been updated in a recent PR: https://github.com/starkware-libs/blockifier/commit/0bdd300140b21fd0a8accb51621341d7e668a5a9